metricflow/test/fixtures/setup_fixtures.py

```diff
@@ -76,7 +80,7 @@ class MetricFlowTestEnvironmentVariables:
 def mf_test_session_state(  # noqa: D
     request: FixtureRequest,
     source_table_snapshot_repository: SqlTableSnapshotRepository,
-) -> MetricFlowTestSessionState:
+) -> Generator[MetricFlowTestSessionState, None, None]:
```

Comment:
I'm very confused as to why this change "just works" - seems to me every call to this fixture should break since you were getting a MetricFlowTestSessionState and now you can't access it without an iterator call.

Is there some Python generator magic I'm missing here?  Or is this pytest fixture + generator magic?